### PR TITLE
Allow arbitrary aspect ratio in IFSuperResolutionPipeline

### PR DESCRIPTION
**Pull request by devxpy**
Fixes https://github.com/huggingface/diffusers/issues/3289

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Thanks a lot for the PR @devxpy, could we add one dummy model test and one big model test for this?

**Comment by patrickvonplaten** [comment]
@williamberman can you take a look here?

File: src/diffusers/pipelines/deepfloyd_if/pipeline_if_superresolution.py [hunk 1]
```diff
@@ -806,8 +806,9 @@ def __call__(
 
         # 2. Define call parameters
 
-        height = self.unet.config.sample_size
-        width = self.unet.config.sample_size
+        height, width = image.shape[2:]
+        width *= 4
+        height *= 4
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/deepfloyd_if/pipeline_if_superresolution.py:811]: Can we get the *4 anywhere from the config? cc @williamberman

**Comment by devxpy** [comment]
I was also a little worried about what would happen to `img.shape` if someone passed in a PIL image instead of an array

**Comment by patrickvonplaten** [comment]
> I was also a little worried about what would happen to `img.shape` if someone passed in a PIL image instead of an array

Ah yeah good point, we need to make sure that it works for all inputs as well :-)

**Comment by AmericanPresidentJimmyCarter** [comment]
For myself I just passed in `height` and `width` as kwargs.

**Comment by Gitterman69** [comment]
How do i use custom ratios?

File: src/diffusers/pipelines/deepfloyd_if/pipeline_if_superresolution.py (hunk 1 above)
Comment (williamberman) [reply src/diffusers/pipelines/deepfloyd_if/pipeline_if_superresolution.py:811]: hmm, I'm not sure if this is right actually

**Comment by williamberman** [comment]
hey @devxpy appreciate the PR!

I think it would be better if we just allow the user to override the height and width through the arguments to __call__

Could you make the change like so? Also adding doc strings to function and adding the arguments to the other IF super resolution pipelines? 

Thanks!

```
diff --git a/src/diffusers/pipelines/deepfloyd_if/pipeline_if_superresolution.py b/src/diffusers/pipelines/deepfloyd_if/pipeline_if_superresolution.py
index 1ba8f888..6faea0d6 100644
--- a/src/diffusers/pipelines/deepfloyd_if/pipeline_if_superresolution.py
+++ b/src/diffusers/pipelines/deepfloyd_if/pipeline_if_superresolution.py
@@ -712,6 +712,8 @@ class IFSuperResolutionPipeline(DiffusionPipeline):
         cross_attention_kwargs: Optional[Dict[str, Any]] = None,
         noise_level: int = 250,
         clean_caption: bool = True,
+        height=None,
+        width=None,
     ):
         """
         Function invoked when calling the pipeline for generation.
@@ -806,8 +808,8 @@ class IFSuperResolutionPipeline(DiffusionPipeline):
 
         # 2. Define call parameters
 
-        height = self.unet.config.sample_size
-        width = self.unet.config.sample_size
+        height = height or self.unet.config.sample_size
+        width = width or self.unet.config.sample_size
 
         device = self._execution_device
```

**Comment by devxpy** [comment]
I wonder if we should have height/width on `IFImg2ImgSuperResolutionPipeline`, since historically, `StableDiffusionImg2ImgPipeline` & `StableDiffusionUpscalePipeline` have been dependent on the input image shape only.

**Comment by patrickvonplaten** [comment]
Gentle nudge @williamberman here

**Review by williamberman — APPROVED** [review]
nice!

**Pull request merged**